fbw-a32nx/src/systems/instruments/src/MCDU/legacy_pages/A320_Neo_CDU_LateralRevisionPage.ts

```diff
@@ -43,8 +44,22 @@ export class CDULateralRevisionPage {
 
     const isPpos = leg === undefined || (legIndexFP === 0 && leg !== targetPlan.originLeg);
     const isFrom = legIndexFP === targetPlan.fromLegIndex && forPlan === FlightPlanIndex.Active && !inAlternate;
-    const isDeparture = legIndexFP === targetPlan.originLegIndex && !isPpos; // TODO this is bogus... compare icaos
-    const isDestination = legIndexFP === targetPlan.destinationLegIndex && !isPpos; // TODO this is bogus... compare icaos
+    const legWaypoint = !isPpos ? mcdu.flightPlanService.active.legElementAt(legIndexFP).definition.waypoint : null;
+    const departure = mcdu.flightPlanService.active.originAirport;
+    const arrival = mcdu.flightPlanService.active.destinationAirport;
+    const currentWaypointAirportIdent = (legWaypoint as Airport)?.airportIdent;
```

Comment:
We should really use a type-guard and properly do this check rather than casting.